src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -2804,11 +2807,20 @@ private function pushChangesToStagingArea($id) {
       $ref_list[] = $ref['commit'].':'.$ref['ref'];
     }
 
-    $err = phutil_passthru(
-      'git push %Ls -- %s %Ls',
-      $push_flags,
-      $staging_uri,
-      $ref_list);
+    // phutil_passthru wraps the given string with single quotes
+    // therefore, "GIT_TRACE=1" is not passed through %s in string formatting
+    if ($this->getArgument('trace-git-push'))
```

Comment:
I think we can use a single format string with no `if` if we just pass in an empty string if the flag is not set. Something link `$environment_vars`.